ui-tests/src/test/java/io/syndesis/qe/pages/apps/todo/Todo.java

```diff
@@ -1,13 +1,53 @@
 package io.syndesis.qe.pages.apps.todo;
 
-import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$;
 import static com.codeborne.selenide.Selenide.$$;
 
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.Selenide;
+import com.codeborne.selenide.SelenideElement;
+import org.assertj.core.api.Assertions;
 import org.openqa.selenium.By;
 
 public class Todo {
 
+    public void refresh(){
+        Selenide.refresh();
+    }
+
     public int getListItemsCount() {
-        return $$(By.xpath("//li[@class='list-group-item']//button[@value='check']")).size();
+        return getListItems().size();
+    }
+
+    public ElementsCollection getListItems(){
+        return $$(By.xpath("//li[@class='list-group-item']//input[@name='task' and @value!='']"));
+    }
+
+    public void setJmsForm(String text){
+        showFormIsNecessary();
+        SelenideElement textArea = $(By.name("message")).shouldBe(Condition.visible);
+        textArea.clear();
+        textArea.sendKeys(text);
+    }
+
+    public void sentJmsMessage(){
+        showFormIsNecessary();
+        SelenideElement button = $(By.name("amq")).shouldBe(Condition.visible);
+        button.click();
+        Assertions.assertThat($(By.xpath("//*[@id=\"jmsToggle\"]/div/ul/li/div/span")).text()).contains("Message sent successfully");
```

Comment:
It might be more helpful to also provide assertion message.

Reply:
In my point of view, it will be helpful in case 
```
$(By.xpath("//*[@id=\"jmsToggle\"]/div/ul/li/div/span")).text()
```
returns message for error case. e.g. _Message was not sent successfully._ 
But now, when the message is not sent successfully, on the screen is shown just ugly PHP stack trace.
e.g.

```
Failed to send message: Stomp\Exception\ConnectionException: Was not possible to read data from stream. (Host: broker-amq-tcp) in /opt/app-root/src/vendor/stomp-php/stomp-php/src/Network/Connection.php:435 Stack trace: #0 /opt/app-root/src/vendor/stomp-php/stomp-php/src/Client.php(237): Stomp\Network\Connection->readFrame() #1 /opt/app-root/src/vendor/stomp-php/stomp-php/src/Client.php(212): Stomp\Client->getConnectedFrame() #2 /opt/app-root/src/index.php(89): Stomp\Client->connect() #3 {main}
```

Reply:
Ah, I am not sure we understand each other. I meant a general message like: "Unexpected response" or sth. like that. That way you get both assertion message and the results of comparison. (In case that the values are messy, the message just improves readability.)